tests/test_json_to_parquet.py

```diff
@@ -0,0 +1,531 @@
+        assert len(nested_table.schema().fields) == 4
+
+    def test_has_nested_fields(self, glue_database_name, glue_flat_table_name,
+                               glue_nested_table_name, glue_context):
+            flat_table = json_to_parquet.get_table(
+                    table_name=glue_flat_table_name,
+                    database_name=glue_database_name,
+                    glue_context=glue_context
+            )
+            flat_table_schema = flat_table.schema()
+            assert not json_to_parquet.has_nested_fields(flat_table_schema)
+            nested_table = json_to_parquet.get_table(
+                    table_name=glue_nested_table_name,
+                    database_name=glue_database_name,
+                    glue_context=glue_context
+            )
+            nested_table_schema = nested_table.schema()
+            assert json_to_parquet.has_nested_fields(nested_table_schema)
+
+    def test_add_index_to_table(self, glue_database_name, glue_database_path,
+                                glue_nested_table_name, glue_context):
+        nested_table = json_to_parquet.get_table(
+                table_name=glue_nested_table_name,
+                database_name=glue_database_name,
+                glue_context=glue_context
+        )
+        nested_table_relationalized = nested_table.relationalize(
+            root_table_name = glue_nested_table_name,
+            staging_path = os.path.join(
+                glue_database_path,
+                "tmp/"
+            )
+        )
+        tables_with_index = {}
+        tables_with_index[glue_nested_table_name] = json_to_parquet.add_index_to_table(
+                table_key=glue_nested_table_name,
+                table_name=glue_nested_table_name,
+                processed_tables=tables_with_index,
+                unprocessed_tables=nested_table_relationalized,
+                glue_context=glue_context
+        )
+        assert (
+                set(tables_with_index[glue_nested_table_name].schema.fieldNames()) ==
+                set(["GlobalKey", "ArrayOfObjectsField", "ObjectField_filename",
+                     "ObjectField_timestamp", "export_end_date"])
+        )
+        table_key = f"{glue_nested_table_name}_ArrayOfObjectsField"
+        tables_with_index[table_key] =  json_to_parquet.add_index_to_table(
+                table_key=table_key,
+                table_name=glue_nested_table_name,
+                processed_tables=tables_with_index,
+                unprocessed_tables=nested_table_relationalized,
+                glue_context=glue_context
+        )
+        assert (
+                set(tables_with_index[table_key].schema.fieldNames()) ==
+                set(['id', 'index', 'filename', 'timestamp', 'GlobalKey'])
+        )
+        child_table_with_index = (tables_with_index[table_key]
+              .toPandas()
+              .sort_values("GlobalKey")
+              .reset_index(drop=True)
+              .drop(columns=["id", "index"]))
```

Comment:
You're doing some operations here, is there a reason we wouldn't want to create a spark dataframe to compare in the assertion?

(I'm ok with this, was just curious)

Reply:
I'm more familiar with doing this in pandas than in pyspark. It's not really clear to me if pyspark has the dataframe manipulation methods needed to directly compare two dataframes since it's not designed for that kind of thing.

Reply:
It's relatively a new thing but while researching for ETL-430 (which you'll see soon), I found out that `pyspark` has a `pandas on spark` object where they mimic the pandas API but for spark dataframes to make it easier for pandas users to scale up to spark without having to learn all of the spark syntax from the ground up. It used to be a standalone project called [koalas](https://koalas.readthedocs.io/en/latest/?badge=latest) but spark saw the benefits and decided to officially include it in `pyspark`.

See here for more info/a sneak peek:

- [Pandas on spark quickstart](https://spark.apache.org/docs/latest/api/python/reference/pyspark.pandas/index.html)
- [Pandas on spark API](https://spark.apache.org/docs/latest/api/python/reference/pyspark.pandas/index.html)